From 9896131da402f36ba83bfbf5f54c1abbe948fd27 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Tue, 29 Mar 2016 19:33:26 +0200
Subject: [PATCH 05/13] nm: replace libgnomekeyring with libsecret

The former is deprecated and the newer API is nicer anyway.
---
 auth-dialog/Makefile.am |  6 ++----
 auth-dialog/main.c      | 39 ++++++++-------------------------------
 configure.ac            |  2 +-
 3 files changed, 11 insertions(+), 36 deletions(-)

diff --git a/auth-dialog/Makefile.am b/auth-dialog/Makefile.am
index 6d3b3d6..04ad7f8 100644
--- a/auth-dialog/Makefile.am
+++ b/auth-dialog/Makefile.am
@@ -3,7 +3,7 @@ libexec_PROGRAMS = nm-strongswan-auth-dialog
 nm_strongswan_auth_dialog_CPPFLAGS = \
 	$(GTK_CFLAGS) \
 	$(LIBGNOMEUI_CFLAGS) \
-	$(GNOMEKEYRING_CFLAGS) \
+	$(LIBSECRET_CFLAGS) \
 	$(NETWORK_MANAGER_CFLAGS) \
 	$(NM_UTILS_CFLAGS) \
 	-DG_DISABLE_DEPRECATED \
@@ -16,7 +16,5 @@ nm_strongswan_auth_dialog_SOURCES =	\
 
 nm_strongswan_auth_dialog_LDADD = \
 	$(GTK_LIBS) \
-	$(LIBGNOMEUI_LIBS) \
-	$(GNOMEKEYRING_LIBS) \
+	$(LIBSECRET_LIBS) \
 	$(NM_UTILS_LIBS)
-
diff --git a/auth-dialog/main.c b/auth-dialog/main.c
index b9fd886..33ed4ee 100644
--- a/auth-dialog/main.c
+++ b/auth-dialog/main.c
@@ -22,7 +22,7 @@
 #include <string.h>
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
-#include <gnome-keyring.h>
+#include <libsecret/secret.h>
 #include <libgnomeui/libgnomeui.h>
 #include <nm-vpn-plugin.h>
 #include <nm-setting-vpn.h>
@@ -32,35 +32,6 @@
 #define NM_DBUS_SERVICE_STRONGSWAN	"org.freedesktop.NetworkManager.strongswan"
 
 /**
- * lookup a password in the keyring
- */
-static char *lookup_password(char *name, char *service)
-{
-	GList *list;
-	GList *iter;
-	char *pass = NULL;
-
-	if (gnome_keyring_find_network_password_sync(g_get_user_name(), NULL, name,
-			NULL, service, NULL, 0, &list) != GNOME_KEYRING_RESULT_OK)
-	{
-		return NULL;
-	}
-
-	for (iter = list; iter; iter = iter->next)
-	{
-		GnomeKeyringNetworkPasswordData *data = iter->data;
-
-		if (strcmp(data->object, "password") == 0 && data->password)
-		{
-			pass = g_strdup(data->password);
-			break;
-		}
-	}
-	gnome_keyring_network_password_list_free(list);
-	return pass;
-}
-
-/**
  * Wait for quit input
  */
 static void wait_for_quit (void)
@@ -162,7 +133,12 @@ int main (int argc, char *argv[])
 	if (!strcmp(type, "eap") || !strcmp(type, "key") || !strcmp(type, "psk") ||
 		!strcmp(type, "smartcard"))
 	{
-		pass = lookup_password(name, service);
+		pass = secret_password_lookup_sync(SECRET_SCHEMA_COMPAT_NETWORK, NULL, NULL,
+						   "user", g_get_user_name(),
+						   "server", name,
+						   "protocol", service,
+						   NULL);
+
 		if ((!pass || retry) && allow_interaction)
 		{
 			if (!strcmp(type, "eap"))
@@ -231,6 +207,7 @@ too_short_retry:
 		if (pass)
 		{
 			printf("password\n%s\n", pass);
+			g_free(pass);
 		}
 	}
 	else
diff --git a/configure.ac b/configure.ac
index 71fcbf2..07811b0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -44,7 +44,7 @@ AM_GLIB_GNU_GETTEXT
 
 PKG_CHECK_MODULES(GTK, gtk+-2.0 >= 2.6)
 PKG_CHECK_MODULES(LIBGNOMEUI, libgnomeui-2.0)
-PKG_CHECK_MODULES(GNOMEKEYRING, gnome-keyring-1)
+PKG_CHECK_MODULES(LIBSECRET, libsecret-1)
 PKG_CHECK_EXISTS([libnm-glib],
 	[PKG_CHECK_MODULES(NM_UTILS, NetworkManager >= 0.9.0 libnm-util libnm-glib libnm-glib-vpn)],
 	[PKG_CHECK_MODULES(NM_UTILS, NetworkManager >= 0.9.0 libnm-util libnm_glib libnm_glib_vpn)]
-- 
2.5.0

